Code/GraphMol/SynthonSpaceSearch/SynthonSpace.h

```diff
@@ -0,0 +1,196 @@
+//
+// Copyright (C) David Cosgrove 2024.
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
```

Comment:
Please add a comment at the top of this that the API here is experimental and may change.
An example of what this might look like is here:
https://github.com/rdkit/rdkit/blob/master/Code/DataStructs/FPBReader.h#L13